de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -198,6 +205,66 @@ protected SWTBotTreeItem openEditor(SWTBotTreeItem item) {
 		return newItem;
 	}
 	
+	/**
+	 * We can only drop tree items to canvas elements. SWTBotGefViewer's canvas element is not 
+	 * capable of beeing accessed directly from external classes though. Further more SWTBotGefViewer's 
+	 * canvas member gets populated during runtime only. Hence we use reflections to get a hold on it here.
+	 * 
+	 * @param item Tree item that is beeing dragged
+	 * @param diagramEditor Graphiti diagram editor which the tree item is beeing dragged onto
+	 */
+	protected void dragTreeItemToDiagramEditor(SWTBotTreeItem item, SWTBotGefEditor diagramEditor) {
+		SWTBotGefViewer viewer = diagramEditor.getSWTBotGefViewer();
+		SWTBotGefFigureCanvas canvas = null;		
+
+		for (Field f : viewer.getClass().getDeclaredFields()) {
+		    if (SWTBOT_CANVAS_FIELD_REFLECTION_NAME.equals(f.getName())) {
+		        f.setAccessible(true);
+		        try {
+		            canvas = (SWTBotGefFigureCanvas) f.get(viewer);
+		        } catch (IllegalArgumentException | IllegalAccessException e) {
+		        	Activator.getDefault().getLog().log(new Status(Status.ERROR, Activator.getPluginId(), Status.ERROR, "Can not access SWTBotGefViewer element or do a proper cast to canvas type", e));

```

Comment:
Whitespacing looks incorrect here.

Reply:
It seems my IDE mixes up tabs and spaces formating regularly, which is hard to spot if displaying whitespaces isnt turned on. Can we add this check to checkstyle globally?

Reply:
I think it is possible. We can look into it.